src/lib/utils/paths.ts

```diff
@@ -0,0 +1,17 @@
+import * as Path from 'path';
+import { isArray } from 'util';
+import { Minimatch, IMinimatch } from 'minimatch';
+
+function createMinimatch(pattern: string): IMinimatch {
+  if (pattern[0] === '.' || pattern[0] === '/' || /^\w+(?!:)($|[/\\])/.test(pattern)) {
+    pattern = Path.resolve(pattern);
+  }
+
+  return new Minimatch(pattern.replace(/[\\]/g, '/'), { dot: true });
+}
+
+export function pathToMinimatch(pattern: string | string[]): IMinimatch | IMinimatch[] {
```

Comment:
Is this function really needed? From the usage I saw, it seems that `createMinimatch` would be sufficient. The overloading of a parameter that is sometimes an array means that you frequently need to cast the result.

I'd prefer to inline the `arr.map(createMinimatch)` in files.

Reply:
It is true that the two methods could be combined. I separated them in an attempt to make the code less bulky and easier to read. But if you prefer to combine them I can. It is also a bit slower to recreate a function inside a function, so there is also a minor performance gain.

Reply:
regarding the `pattern: string | string[]` is because the returned path can be one or several paths. When only one path is given it is just a string otherwise an array of strings is given, hence the double type definition.